### PR TITLE
Link the send loop with the edus contexts

### PR DESCRIPTION
**Pull request by JorikSchellekens**
The contexts were being filtered too early so the send loop wasn't
being linked to them unless the destination was whitelisted.

This also fixes a bug where the json encoding was tripping up over a None value.

### CODE REVIEW

File: changelog.d/5984.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Link send loop opentracing to edu contexts regardles of destination.
```
Comment (erikjohnston) [inline changelog.d/5984.bugfix]: Bug fix changelogs should really be in the form of `Fix ...`

```suggestion
Fix sending of EDUs when opentracing is enabled with an empty whitelist.
```

**Comment by erikjohnston** [comment]
Though you probably want to see whats up with the tests....

File: changelog.d/5984.bugfix (hunk 1 above)
Comment (JorikSchellekens) [reply changelog.d/5984.bugfix]: Is there documentation for the format of a changelog?

Comment (erikjohnston) [reply changelog.d/5984.bugfix]: Not particularly, and its not a hard and fast rule, its more about having the changelog describe whats being fixed (or added, etc)

File: synapse/federation/sender/transaction_manager.py [hunk 2]
```diff
@@ -59,9 +60,13 @@ def send_new_transaction(self, destination, pending_pdus, pending_edus):
...
         # opentracing is disabled. (Yay speed!)
 
-        span_contexts = (
-            extract_text_map(json.loads(edu.get_context())) for edu in pending_edus
-        )
+        span_contexts = []
+        keep_destination = whitelisted_homeserver(destination)
+
+        for edu in pending_edus:
+            span_contexts.append(extract_text_map(json.loads(edu.get_context())))
```
Comment (erikjohnston) [inline synapse/federation/sender/transaction_manager.py]:
```suggestion
            context = edu.get_context()
            if context:
                span_contexts.append(extract_text_map(json.loads(context)))
```

Comment (erikjohnston) [reply synapse/federation/sender/transaction_manager.py]: Or some such :)

**Pull request merged**